packages/edit-site/src/components/sync-state-with-url/use-sync-path-with-url.js

```diff
@@ -109,6 +109,13 @@ export default function useSyncPathWithURL() {
 	}, [ navigatorLocation?.path, navigatorParams, history ] );
 
 	useEffect( () => {
+		if (
+			Object.entries( urlParams ).every( ( [ key, value ] ) => {
+				return currentUrlParams.current[ key ] === value;
+			} )
+		) {
+			return;
+		}
```

Comment:
This fixes a bug where `goTo` is called twice as a result of the call to `updateUrlParams` in the effect above. `goTo` should only ever be called once either as a result of pressing a navigation button or changing the URL.

Reply:
I think what @andrewserong [noticed](https://github.com/WordPress/gutenberg/pull/52456#pullrequestreview-1521285940) is caused by this check. 

From what I can see, `currentUrlParams` and `urlParams` are the same, e.g., `{postType: 'page', postId: '27'}`, when returning to the same page after going back.